pkg/spdk/engine.go

```diff
@@ -1047,8 +1055,15 @@ func (e *Engine) ReplicaDelete(spdkClient *spdkclient.Client, replicaName, repli
 		return fmt.Errorf("replica %s recorded address %s does not match the input address %s for engine %s replica delete", replicaName, e.ReplicaAddressMap[replicaName], replicaAddress, e.Name)
 	}
 
+	e.log.Infof("Removing base bdev %v from engine", e.ReplicaBdevNameMap[replicaName])
 	if _, err := spdkClient.BdevRaidRemoveBaseBdev(e.ReplicaBdevNameMap[replicaName]); err != nil && !jsonrpc.IsJSONRPCRespErrorNoSuchDevice(err) {
-		return err
+		return errors.Wrapf(err, "failed to remove base bdev %s for deleting replica %s", e.ReplicaBdevNameMap[replicaName], replicaName)
+	}
+
+	controllerName := helperutil.GetNvmeControllerNameFromNamespaceName(replicaName)
+	e.log.Infof("Detaching controller %v", controllerName)
+	if _, err := spdkClient.BdevNvmeDetachController(controllerName); err != nil && !jsonrpc.IsJSONRPCRespErrorNoSuchDevice(err) {
+		return errors.Wrapf(err, "failed to detach controller %s for deleting replica %s", controllerName, replicaName)
```

Comment:
Shouldn't we suspend the dm device and restart the rest of the frontend before doing this?

Reply:
No, we don't need to suspend the dm device.
If one of replicas is being deleted, the IO will be queued at SPDK raid layer. After deletion, the IO continues.

Reply:
OK, got it. But it's better to check if the replica is the local lvol before`BdevNvmeDetachController`. 
Besides, the log message should be:
```
Detaching the corresponding NVMf controller %v during remote replica %s delete
```

Reply:
Addressed. Thank you.